server/src/main/java/org/opensearch/transport/BaseInboundMessage.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.transport;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Base class for inbound data as a message.
+ * Different implementations are used for different protocols.
+ *
+ * @opensearch.internal
+ */
+@ExperimentalApi
+public interface BaseInboundMessage {
+
+    /**
+     * The protocol used to encode this message
+     */
+    static String NATIVE_PROTOCOL = "native";
+    static String PROTOBUF_PROTOCOL = "protobuf";
+
+    /**
+     * @return the protocol used to encode this message
+     */
+    public String getProtocol();
```

Comment:
```suggestion
default  public String getProtocol() {
    return "native";
}
```

It changes public API (`InboundMessage`) so we need to be careful (if we want to backport to `2.x`)

Reply:
I would facade this from the interface we want.

```java
@Deprecated
InboundMessage < a.b.c.d.InboundMessageInterfaceWeWant
    public String getProtocol() {
       ...
    }

    public void setProtocol(String value) {
         // throws or checks or something else
    }
}
```